### ISSUE TITLE
BUG: MessageContent.FromImageUrl(Uri) incompatible with "data url"

### ISSUE THREAD

**Issue by crickman** [openai/openai-dotnet#135]
Open uses a custom URL format for providing an image message where the base64 encoded image bytes are provided:

`data:image/jpeg;base64,{base64_image}`

This format is not accepted by the .Net Uri constructor.

https://platform.openai.com/docs/guides/vision/uploading-base-64-encoded-images

**Comment by joseharriaga** [comment]
Thank you for reaching out, @crickman ! The documentation that you're linking to refers to the Chat Completions API, while the `MessageContent` class that you mention is part of the Assistants API instead. For Chat Completions, we have an example here of how to provide an image as `BinaryData` as part of the content of a `UserChatMessage`: 
🔗 https://github.com/openai/openai-dotnet/blob/main/examples/Chat/Example05_ChatWithVision.cs

**Comment by sibaji-vayu** [comment]
@joseharriaga  I'm facing the same issue with OpenAI Assistant. I'm unable to post base64-encoded binary image data using the .NET SDK. Any update on this?

**Comment by jsquire** [comment]
@KrzysztofCwalina, @christothes:  This looks to be a legitimate blocker for some scenarios.   It has been resolved in .NET 10 _(see: [Uri length limits removed](https://learn.microsoft.com/en-us/dotnet/core/compatibility/networking/10.0/uri-length-limits-removed))_.  Given the pending .NET 10 release in November,  I'd appreciate thoughts on whether this is something that we should address by considering a move to `string` image data URLs or we would advise using .NET 10 betas in the interim.

**Comment by KrzysztofCwalina** [comment]
I just confirmed that it's possible to create Uri instances with large (image size) data in .NET 10. So I would prefer to just recommend users use .NET 10 if they want to use data URIs.

**Comment by stephentoub** [comment]
> I just confirmed that it's possible to create Uri instances with large (image size) data in .NET 10

Yup, in https://github.com/dotnet/runtime/pull/117287 we removed the long-standing limitation.
